### ISSUE TITLE
Google Cloud Functions - update in place on demand

### ISSUE THREAD

**Issue by szwalkowski** [hashicorp/terraform-provider-google#14965]
Hi

I want my functions to be updated in place on demand.
Currently we are having force update by trick with label that has dynamic timestamp value.

I've tried replace_triggered_by but it's forcing replace, so function is going to be deleted and launched again (it's out of option, cause it's downtime).

Is there anyway that we can send something from configuration (like flag to force update) that would work as desired?

The cheesy way we have with that timestamp that is switching with static text if such flag is off. But that's at least one more deploy - and we want to avoid it. 

Is that possible?
I was trying to find this via Terraform itself, but it's just not possible: https://github.com/hashicorp/terraform/issues/33399

**Comment by edwardmedia** [comment]
@szwalkowski are you able to do such `update` with other tools like gcloud or console?

**Comment by Boardtale** [comment]
@edwardmedia yeah ;) This is just simple 'edit' on google cloud function. When you change env variable, memory allocation etc. Or add label as we do everytime with dynamic label in terraform.

But I want to do this 'on demand'. Please edit (with no changes) and deploy function.
I can do this with fake label that is being change everytime, but that's the problem actually also sometimes.

And I cannot do this with 'replace_triggered_by' or -replace=google_cloud_function.my_function cause whole function is going to be deleted first.

**Comment by edwardmedia** [comment]
b/288706163
